controllers/vrg_volrep.go

```diff
@@ -1466,6 +1466,18 @@ func setPVCClusterDataProtectedCondition(protectedPVC *ramendrv1alpha1.Protected
 	}
 }
 
+// ensureVRDeletedFromAPIServer adds an additional step to ensure that we wait for volumereplication deletion
+// from API server before moving ahead vrg finalizer removal.
+func (v *VRGInstance) ensureVRDeletedFromAPIServer(vrNamespacedName types.NamespacedName) error {
+	if err := v.reconciler.APIReader.Get(v.ctx, vrNamespacedName, v.instance); err != nil {
+		v.log.Info("Wait to delete VolumeReplication resource from API server", "Error", err)
+
+		return err
+	}
+
+	return nil
```

Comment:
Ensure err is NotFound, we do not want to report resource as being delete due to other Get errors.

Reply:
If err is non-nil and not NotFound function still returns nil. It should return an err.